testonly/hammer/hammer.go

```diff
@@ -165,8 +166,26 @@ func (c MapConfig) String() string {
 		c.MapID, c.EPBias, c.Operations, c.EmitInterval, c.RetryErrors)
 }
 
+// SetULimit calls `ulimit -n` to set the maximum number of open files.
+// See https://wilsonmar.github.io/maximum-limits/
+func SetULimit(uLimit uint64) error {
+	var rLimit syscall.Rlimit
+	err := syscall.Getrlimit(syscall.RLIMIT_NOFILE, &rLimit)
+	if err != nil {
+		return err
+	}
+	if uLimit > rLimit.Max {
+		return fmt.Errorf("`ulimit -n %v -S` failed because it is greater than the hard limit %v", uLimit, rLimit.Max)
```

Comment:
Maybe change the error string - this implies that the reader knows the flags to the `ulimit` binary, but also might confuse them since it doesn't _actually_ run the `ulimit` binary...